node/actors/os_procs.go

```diff
@@ -60,23 +61,35 @@ func (proc *OsProcess) Run() error {
 		}
 		return err
 	}
+
+	proc.logger.Debug("OS process terminated successfully", slog.Int("pid", proc.cmd.Process.Pid))
+	proc.cmd = nil
+
+	return nil
+}
+
+// Sends an interrupt signal to the running process (if it's running)
+func (proc *OsProcess) Stop(reason string) error {
 	if proc.cmd.Process != nil {
-		proc.logger.Debug("OS process terminated successfully", slog.Int("pid", proc.cmd.Process.Pid))
+		return proc.cmd.Process.Signal(os.Interrupt)
 	}
-
 	return nil
 }
 
-// Stops the running process (if it's running). Note that this sends
+// Kills the running process (if it's running). Note that this sends
 // a signal and then exits, which means the actual OS process might stop
 // "some time" later
-func (proc *OsProcess) Stop(reason string) error {
+func (proc *OsProcess) Kill() error {
 	if proc.cmd.Process != nil {
 		return proc.cmd.Process.Kill()
 	}
 	return nil
 }
 
+func (proc *OsProcess) IsRunning() bool {
+	return proc.cmd != nil
```

Comment:
might also want to add a check for proc.cmd.Process != nil

Reply:
So I found that even after the stop, the proc.cmd.Process was never nil...thats why I manually set `proc.cmd = nil` after cmd.Wait unblocks

Reply:
Huh. Learn something new every day